telegram/ext/jobqueue.py

```diff
@@ -70,30 +71,25 @@ def __init__(self):
     def set_dispatcher(self, dispatcher):
         self._dispatcher = dispatcher
 
-    def _put(self, job, next_t=None, last_t=None):
-        if next_t is None:
-            next_t = job.interval
-            if next_t is None:
-                raise ValueError('next_t is None')
-
-        if isinstance(next_t, datetime.datetime):
-            next_t = (next_t - datetime.datetime.now()).total_seconds()
-
-        elif isinstance(next_t, datetime.time):
-            next_datetime = datetime.datetime.combine(datetime.date.today(), next_t)
-
-            if datetime.datetime.now().time() > next_t:
-                next_datetime += datetime.timedelta(days=1)
-
-            next_t = (next_datetime - datetime.datetime.now()).total_seconds()
+    def _put(self, job, next_t=None, previous_t=None):
+        """
+        Enqueues the job, scheduling its next run at the correct time.
 
-        elif isinstance(next_t, datetime.timedelta):
-            next_t = next_t.total_seconds()
+        Args:
+            job (telegram.ext.Job): job to enqueue
+            next_t (optional):
+                Time for which the job should be scheduled. The precise semantics of this parameter
+                depend on its type (see :func:`telegram.ext.JobQueue.run_repeating` for details).
+                Defaults to now + ``job.interval``.
+            previous_t (optional):
+                Time at which the job last ran (``None`` if it hasn't run yet).
+        """
 
-        next_t += last_t or time.time()
+        # get time at which to run:
+        next_t = to_float_timestamp(next_t or job.interval, reference_timestamp=previous_t)
```

Comment:
Actually, I've come to think that if both `next_t` and `job_interval` are `None`, we should raise an error, since we have no idea when to schedule this job. We cannot enqueue a job with priority (time) `None`—it'll raise an obscure (from the perspective of `_put`'s caller) error about `None` not being comparable to `float` if the queue is not empty, or if it gets past this, the [comparison between `t` and `now` in `JobQueue.tick()`](https://github.com/python-telegram-bot/python-telegram-bot/blob/bbcff96804bc8dc40ad948dabbce264a7f83538c/telegram/ext/jobqueue.py#L258) will break for the same reason.